lxd/container_lxc.go

```diff
@@ -160,6 +160,24 @@ func lxcSetConfigItem(c *lxc.Container, key string, value string) error {
 		}
 	}
 
+	if key == "lxc.console.logsize" {
+		if !util.RuntimeLiblxcVersionAtLeast(3, 0, 0) {
+			return fmt.Errorf(`The console ringbuffer options require libxc >= 3.0`)
+		}
+	}
+
+	if key == "lxc.console.rotate" {
+		if !util.RuntimeLiblxcVersionAtLeast(3, 0, 0) {
+			return fmt.Errorf(`Rotating the console logfile requires libxc >= 3.0`)
+		}
+	}
+
+	if strings.HasPrefix(key, "lxc.console.buffer.") {
+		if !util.RuntimeLiblxcVersionAtLeast(3, 0, 0) {
+			return fmt.Errorf(`The console ringbuffer options require libxc >= 3.0`)
+		}
+	}
```

Comment:
Why do we need this? I don't think we've done anything like that for other keys and it seems a bit odd to maintain a complete list of new keys and liblxc versions inside of LXD.

LXD should have the right version checks when it sets something itself but if the user sets a raw.lxc that's not supported, then it's on them.

Reply:
Uhm, aren't these going to be accessible by the user through LXD config keys? That was my last impression from discussions we had. In that case we need to do something similar which we do for `lxc.prlimit.` just slightly above.